src/System.Management.Automation/utils/ClrFacade.cs

```diff
@@ -555,18 +559,33 @@ internal static Encoding GetOEMEncoding()
         {
             if (s_oemEncoding == null)
             {
-#if CORECLR     // The OEM code page '437' is not supported by CoreCLR.
-                // Use the default encoding (ISO-8859-1, CodePage 28591) as the OEM encoding in OneCore powershell.
+#if UNUX        // PowerShell Core on Unix
                 s_oemEncoding = GetDefaultEncoding();
-#else
+#elif CORECLR   // PowerShell Core on Windows
+                EncodingRegisterProvider()
+
+                uint oemCp = NativeMethods.GetOEMCP();
+                s_oemEncoding = Encoding.GetEncoding((int)oemCp);
+
+#else           // Windows PowerShell
                 uint oemCp = NativeMethods.GetOEMCP();
                 s_oemEncoding = Encoding.GetEncoding((int)oemCp);
 #endif
```

Comment:
I didn't notice the duplicate code between coreclr and fullclr. In this case, maybe the following is better?
```
#if UNIX
         s_oemEncoding = GetDefaultEncoding();
#else
#if CORECLR
         EncodingRegisterProvider()
#endif
         uint oemCp = NativeMethods.GetOEMCP();
         s_oemEncoding = Encoding.GetEncoding((int)oemCp);
#end
```